lib/libevent/libevent.nit

```diff
@@ -458,6 +469,35 @@ extern class ConnectionListener `{ struct evconnlistener * `}
 	end
 end
 
+# A listener acting on an Unix Socket, spawns `Connection` on new connections
+extern class UnixConnectionListener `{ struct evconnlistener * `}
+	super ConnectionListener
+
+	private new bind_to(base: NativeEventBase, address: CString, file: CString, factory: ConnectionFactory)
+	import ConnectionFactory.accept_connection, error_callback `{
+		struct sockaddr_un sun;
+		struct evconnlistener *listener;
+
+		unlink(file);
```

Comment:
If I'm not mistaken, this can be destructive... Imagine the user specifies wrongly a file path instead of the socket path, this would delete his file.

You may want to check if the file exists instead, and fail if it does.

Reply:
The common practice about unix socket is called "unlink before bind". This seems to have the best trade-offs even if I'm not a fan of it.

About the bad user specification, this seems unfortunate but falls in the same issue than `nitc prog.nit -o important_database.db`